nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java

```diff
@@ -0,0 +1,180 @@
+   * NullAway will serialize information and suggests type changes to resolve them, in case these
+   * errors could be fixed by adding a {@code @Nullable} annotation. These type change suggestions
+   * are in form of {@link com.uber.nullaway.fixserialization.out.SuggestedFixInfo} instances and
+   * will be serialized at output directory. If deactivated, no {@code SuggestedFixInfo} will be
+   * created and the output file will remain untouched.
+   */
+  public final boolean suggestEnabled;
+  /**
+   * If activated, serialized information of a fix suggest will also include the enclosing method
+   * and class of the element involved in error. Finding enclosing elements is costly and will only
+   * be computed at request.
+   */
+  public final boolean suggestEnclosing;
+  /** The directory where all files generated/read by Fix Serialization package resides. */
+  public final String outputDirectory;
+
+  public final AnnotationConfig annotationConfig;
+
+  public final Serializer serializer;
+
+  public FixSerializationConfig() {
+    suggestEnabled = false;
+    suggestEnclosing = false;
+    annotationConfig = new AnnotationConfig();
+    outputDirectory = null;
+    serializer = null;
+  }
+
+  public FixSerializationConfig(
+      boolean suggestEnabled,
+      boolean suggestEnclosing,
+      AnnotationConfig annotationConfig,
+      String outputDirectory) {
+    this.suggestEnabled = suggestEnabled;
+    this.suggestEnclosing = suggestEnclosing;
+    this.outputDirectory = outputDirectory;
+    this.annotationConfig = annotationConfig;
+    serializer = new Serializer(this);
+  }
+
+  /**
+   * Sets all flags based on their values in the configuration file.
+   *
+   * @param configFilePath Path to the serialization config file written in xml.
+   */
+  public FixSerializationConfig(String configFilePath) {
+    Preconditions.checkNotNull(configFilePath);
+    Document document;
+    try {
+      DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+      DocumentBuilder builder = factory.newDocumentBuilder();
+      document = builder.parse(Files.newInputStream(Paths.get(configFilePath)));
+      document.normalize();
+    } catch (Exception e) {
+      throw new RuntimeException(
+          "Error in reading/parsing config at path: " + configFilePath + "\n" + e);
+    }
+    this.outputDirectory =
+        XMLUtil.getValueFromTag(document, "serialization:path", String.class).orElse(null);
+    Preconditions.checkNotNull(
+        this.outputDirectory, "Error in FixSerialization Config: Output path cannot be null");
+    suggestEnabled =
+        XMLUtil.getValueFromAttribute(document, "serialization:suggest", "active", Boolean.class)
+            .orElse(false);
+    suggestEnclosing =
+        XMLUtil.getValueFromAttribute(document, "serialization:suggest", "enclosing", Boolean.class)
+                .orElse(false)
+            && suggestEnabled;
```

Comment:
Shouldn't we rather be checking that if active is set to false, but enclosing is set to true, then the configuration file is incorrect and we should throw an error?

Reply:
cf3d6d9e36c281e0782f83ec2fb8764722ec1abf